src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -87,34 +88,86 @@ export const errorMessageKeys = {
 */
 export function validateEmptyFields(
   formData: any,
-  formLayout: any,
+  formLayout: ILayout,
   language: any,
   hiddenFields: string[],
+  repeatingGroups: IRepeatingGroups,
 ) {
   const validations: any = {};
-  formLayout.forEach((component: any) => {
-    if (!hiddenFields.includes(component.id) && component.required) {
-      const fieldKey = Object.keys(component.dataModelBindings)
-        .find((binding: string) => component.dataModelBindings[binding]);
-      const value = formData[component.dataModelBindings[fieldKey]];
-      if (!value && fieldKey) {
-        validations[component.id] = {};
-        const componentValidations: IComponentValidations = {
-          [fieldKey]: {
-            errors: [],
-            warnings: [],
-          },
-        };
-        componentValidations[fieldKey].errors.push(
-          getLanguageFromKey('form_filler.error_required', language),
-        );
-        validations[component.id] = componentValidations;
+  let fieldsInGroup = [];
+  const groupsToCheck = formLayout.filter((component) => component.type === 'group');
+  groupsToCheck.forEach((groupComponent: ILayoutGroup) => {
+    fieldsInGroup = fieldsInGroup.concat(groupComponent.children);
+  });
+  const fieldsToCheck = formLayout.filter((component) => {
+    return (
+      !hiddenFields.includes(component.id)
+        && component.type !== 'group'
```

Comment:
Better perf if this check is done first?

```
component.type !== 'group'
```

Reply:
Might be, changed the order.